rcl_yaml_param_parser/src/parser.c

```diff
@@ -393,6 +393,186 @@ rcl_params_t * rcl_yaml_node_struct_init(
   return params_st;
 }
 
+///
+/// Copy the rcl_params_t parameter structure
+///
+rcl_params_t * rcl_yaml_node_struct_copy(
+  const rcl_params_t * params_st)
+{
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(params_st, NULL);
+
+  rcutils_allocator_t allocator = params_st->allocator;
+  rcl_params_t * out_params_st = rcl_yaml_node_struct_init(allocator);
+
+  if (NULL == out_params_st) {
+    RCUTILS_SAFE_FWRITE_TO_STDERR("Error allocating mem");
+    return NULL;
+  }
+
+  rcutils_ret_t ret;
+  for (size_t node_idx = 0U; node_idx < params_st->num_nodes; node_idx++) {
```

Comment:
nitpick: use preincrement `++node_idx`.

Same below.

Reply:
Done in ac845e9b89c0c50870c652d2276d91d5498b9b11.